dali/base/dadfs.cpp

```diff
@@ -12044,9 +12069,10 @@ IDFAttributesIterator* CDistributedFileDirectory::getLogicalFilesSorted(
         }
     }
     IArrayOf<IPropertyTree> results;
-    Owned<IElementsPager> elementsPager = new CDFUPager(udesc, (const char*) filters, localFilters, (const char*) localFilterBuf,
-        so.length()?so.str():NULL );
-    getElementsPaged(elementsPager,startOffset,maxNum,NULL,"",cacheHint,results,total,false);
+    Owned<IElementsPagerEx> elementsPager = new CDFUPager(udesc, (const char*) filters, localFilters, (const char*) localFilterBuf,
+        so.length()?so.str():NULL);
+    getElementsPaged(elementsPager,startOffset,maxNum,NULL,"",cacheHint,results,NULL,false);
+    *total = elementsPager->getTotalElements();
```

Comment:
Related to getTotalElements() comment above, better to consolidate on 1 way to get totalFiles, instead of passing in NULL here and setting outside.

They can all set inside, if they call elementsPager->getTotalElements() (and all implement of course)